$OpenBSD: patch-lib_common_socket_c,v 1.3 2021/12/03 20:57:47 tb Exp $

Silence compiler warning. The get_session_cb has had const since LibreSSL 2.8.

Index: lib/common/socket.c
--- lib/common/socket.c.orig
+++ lib/common/socket.c
@@ -920,7 +920,7 @@ static void create_ossl(h2o_socket_t *sock)
 }
 
 static SSL_SESSION *on_async_resumption_get(SSL *ssl,
-#if OPENSSL_VERSION_NUMBER >= 0x1010000fL && !defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER >= 0x1010000fL && (!defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER >= 0x2080000fL)
                                             const
 #endif
                                             unsigned char *data,
